sdk/python/kfp/_client.py

```diff
@@ -1401,7 +1401,8 @@ def list_pipeline_versions(
             pipeline_id: str,
             page_token: str = '',
             page_size: int = 10,
-            sort_by: str = ''
+            sort_by: str = '',
+            filter: str = None
```

Comment:
nit: if the default value is `None`, then it should be `:Optional[str] = None`

BTW, were you able to test this with both default `None` and a real value?

Reply:
To be honest I'm just copying the pattern established in [this earlier PR](https://github.com/kubeflow/pipelines/pull/6748) that introduced the `filter` argument 😅  I can certainly confirm it works with the `filter` set as we're running this within the Zillow-specific fork of the `kfp-sdk` we maintain 😄 

I've fixed this type hint (and updated the type hints for all the `list_*` client method arguments too), I'm happy to test out with no filter if needs be. Let me know if you want me to try it out or if you're happy without :)

Reply:
Yes, this is more of a style thing, it doesn't affect runtime behavior. Our code may have many other styling issues, and we want to be more compliance to the style for newly added code at least.
https://google.github.io/styleguide/pyguide.html#3195-nonetype